### PR TITLE
Linalg kron func

### PR DESCRIPTION
**Pull request by Alexander-Makaryev**

### CODE REVIEW

File: dpnp/dpnp_iface_linearalgebra.py [hunk 1]
```diff
@@ -194,21 +194,30 @@ def inner(x1, x2, **kwargs):
     return call_origin(numpy.inner, x1, x2, **kwargs)
 
 
-def kron(x1, x2, **kwargs):
+def kron(a, b):
```
Comment (shssf) [inline dpnp/dpnp_iface_linearalgebra.py:197]: please don't use single letter variables names

File: dpnp/dpnp_algo/dpnp_algo.pyx [hunk 2]
```diff
@@ -98,6 +98,10 @@ cpdef dparray dpnp_array(obj, dtype=None):
     cdef elem_dtype
     cdef dparray_shape_type obj_shape
 
+    # convert scalar to tuple
+    if dpnp.isscalar(obj):
```
Comment (shssf) [inline dpnp/dpnp_algo/dpnp_algo.pyx:102]: maybe it is better to use https://github.com/IntelPython/dpnp/blob/dea4c7a2712bbb1c60b58e7b142cd318ef7a6641/dpnp/dpnp_utils/dpnp_algo_utils.pyx#L296?

**Review by shssf — APPROVED** [review]

File: dpnp/dpnp_iface_linearalgebra.py (hunk 1 above)
Comment (Alexander-Makaryev) [reply dpnp/dpnp_iface_linearalgebra.py:197]: previously we decided to use the same names (as in original numpy) in python interface to be compliant with original numpy and its documentation, that is why I made these changes

File: dpnp/dpnp_algo/dpnp_algo.pyx (hunk 2 above)
Comment (Alexander-Makaryev) [reply dpnp/dpnp_algo/dpnp_algo.pyx:102]: I think that in this place it is better to explicitly show that we convert scalar to tuple tobe able to work with it as with sequence object.
BTW _object_to_tuple(object obj) will not work with all scalars, I made a separate PR for this https://github.com/IntelPython/dpnp/pull/512

**Pull request merged**